index.bs

```diff
@@ -364,11 +400,54 @@ When asked to <dfn export>report the layout shift</dfn> for an active
     1. Set |newEntry|'s <dfn attribute for=LayoutShift>hadRecentInput</dfn>
         attribute to <code>true</code> if {{LayoutShift/lastInputTime}} is less
         than 500 milliseconds in the past, and <code>false</code> otherwise.
+    1. Set |newEntry|'s <dfn attribute for=LayoutShift>sources</dfn> attribute
+        to the result of invoking the algorithm to <a>report the layout shift
+        sources</a> for |D|.
     1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a>
         |newEntry| object.
 
 </div>
 
+Report the layout shift sources {#sec-report-layout-shift-sources}
+------------------------------------------------------------------
+
+<div algorithm="report the layout shift sources">
+When asked to <dfn export>report the layout shift sources</dfn> for an active
+{{Document}} |D|, run the following steps:
+
+1. Let |S| be any set which satisfies these constraints:
+    * Each member of |S| is in the <a>unstable node set</a> of |D|.
+    * |S| contains at least 1 member, and not more than 5 members.
+    * |S| does not contain two members |s1| and |s2| such that the
+        <a>node impact region</a> of |s1| is a subset of the
+        <a>node impact region</a> of |s2|.
+    * The <a>unstable node set</a> of |D| does not contain any member |u|
+        such that the area of the <a>node impact region</a> of |u| is
+        greater than the area of the <a>node impact region</a> of any
+        member of |S|.
```

Comment:
I think the `unstable node set` can include shadow DOM, right? But here we cannot expose that, so we need to add a condition for that in here.

Reply:
Added a constraint for this.